finch/processes/utils.py

```diff
@@ -0,0 +1,39 @@
+import re
+
+import requests
+from siphon.catalog import TDSCatalog
+
+
+def is_opendap_url(url):
+    if url and not url.startswith("file"):
+        r = requests.get(url + ".dds")
+        if r.status_code == 200 and r.content.decode().startswith("Dataset"):
+            return True
+    return False
+
+
+def get_opendap_datasets(catalog_url, variable, rcp):
+    catalog = TDSCatalog(catalog_url)
+
+    urls = []
+
+    for dataset in catalog.datasets.values():
+        if variable in dataset.name and rcp in dataset.name:
+            urls.append(dataset.access_urls["OPENDAP"])
+
+        # using requests (a bit slower, and the Thredds server didn't seem to like it)
```

Comment:
I changed the code a bit, and made the `.das` link parsing actually work. But the default is still to parse the filenames only. The boreas Thredds catalog was getting very unresponsive when I sent it a lot of small requests... and I'm a single client, it would be worse if there were lots of concurrent users. I wouldn't use something else than filename parsing for now, until someone has had time to take a look into this on the Thredds side.